Gateway/Request/RiskDataBuilder.php

```diff
@@ -23,34 +23,88 @@
+use Adyen\Payment\Helper\Config;
+use Magento\Payment\Gateway\Helper\SubjectReader;
 use Magento\Payment\Gateway\Request\BuilderInterface;
+use Magento\Sales\Model\Order\Item;
+use Magento\Store\Model\StoreManagerInterface;
 
 class RiskDataBuilder implements BuilderInterface
 {
     /**
-     * @var Requests
+     * @var ChargedCurrency
      */
-    private $adyenRequestsHelper;
+    private $chargedCurrency;
+    /**
+     * @var Config
+     */
+    private $config;
+    /**
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
 
     /**
      * PaymentDataBuilder constructor.
      *
-     * @param Requests $adyenRequestsHelper
+     * @param Config $config
+     * @param StoreManagerInterface $storeManager
+     * @param ChargedCurrency $chargedCurrency
      */
     public function __construct(
-        Requests $adyenRequestsHelper
+        Config $config,
+        StoreManagerInterface $storeManager,
+        ChargedCurrency $chargedCurrency
     ) {
-        $this->adyenRequestsHelper = $adyenRequestsHelper;
+        $this->chargedCurrency = $chargedCurrency;
+        $this->config = $config;
+        $this->storeManager = $storeManager;
     }
 
     /**
      * @param array $buildSubject
      * @return array
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
      */
     public function build(array $buildSubject)
     {
-        $request['body'] = $this->adyenRequestsHelper->buildRiskData([]);
-        return $request;
+        $requestBody = [];
+        if ($this->config->sendAdditionalRiskData($this->storeManager->getStore()->getId())) {
+            $paymentDataObject = SubjectReader::readPayment($buildSubject);
+            $payment = $paymentDataObject->getPayment();
+            $order = $payment->getOrder();
+            $amountCurrency = $this->chargedCurrency->getOrderAmountCurrency($order);
+            $currencyCode = $amountCurrency->getCurrencyCode();
+
+            $itemIndex = 0;
+            foreach ($order->getItems() as $item) {
+                /** @var Item $item */
+                if ($item->getPrice() == 0 && !empty($item->getParentItem())) {
+                    // products with variants get added to the order twice.
+                    continue;
+                }
+
+                $requestBody['additionalData']['riskdata.basket.item'.$itemIndex.'.amountPerItem'] = $item->getPrice();
+                $requestBody['additionalData']['riskdata.basket.item'.$itemIndex.'.currency'] = $currencyCode;
+                $requestBody['additionalData']['riskdata.basket.item'.$itemIndex.'.itemID'] = $item->getQuoteItemId();
+                $requestBody['additionalData']['riskdata.basket.item'.$itemIndex.'.productTitle'] = $item->getName();
+                $requestBody['additionalData']['riskdata.basket.item'.$itemIndex.'.quantity'] = $item->getQtyOrdered();
+                $requestBody['additionalData']['riskdata.basket.item'.$itemIndex.'.sku'] = $item->getSku();
```

Comment:
We can avoid a bit of duplication like this: 
```suggestion
        $prefix = "riskdata.basket.item$itemIndex.";
        $requestBody['additionalData'] = [
            $prefix . 'amountPerItem' => $item->getPrice(),
            $prefix . 'currency' => $currencyCode,
            $prefix . 'itemID' => $item->getQuoteItemId(),
            $prefix . 'productTitle' => $item->getName(),
            $prefix . 'quantity' => $item->getQtyOrdered(),
            $prefix . 'sku' => $item->getSku(),
        ];
```
The same idea can be used on other places where we populate such fields.

Reply:
efa045e